src/tests/test_api.py

```diff
@@ -148,38 +152,46 @@ def test_stashcache_file(key, endpoint, fqdn, resource_stashcache_files):
             else:
                 assert response.status_code != 200 or not response.data
 
-        resources = client.get('/miscresource/json').json
-        resources_stashcache_files = client.get('/resources/stashcache-files').json
+        try:
+
+            resources = client.get('/miscresource/json').json
+            resources_stashcache_files = client.get('/resources/stashcache-files').json
 
-        # Sanity check: have a reasonable number of resources
-        assert len(resources_stashcache_files) > 20
+            # Sanity check: have a reasonable number of resources
+            assert len(resources_stashcache_files) > 20
 
-        keys_and_endpoints = [
-            ("CacheAuthfilePublic",  "/cache/Authfile-public"),
-            ("CacheAuthfile",        "/cache/Authfile"),
-            ("CacheScitokens",       "/cache/scitokens.conf"),
-            ("OriginAuthfilePublic", "/origin/Authfile-public"),
-            ("OriginAuthfile",       "/origin/Authfile"),
-            ("OriginScitokens",      "/origin/scitokens.conf")
-        ]
+            keys_and_endpoints = [
+                ("CacheAuthfilePublic",  "/cache/Authfile-public"),
+                ("CacheAuthfile",        "/cache/Authfile"),
+                ("CacheScitokens",       "/cache/scitokens.conf"),
+                ("OriginAuthfilePublic", "/origin/Authfile-public"),
+                ("OriginAuthfile",       "/origin/Authfile"),
+                ("OriginScitokens",      "/origin/scitokens.conf")
+            ]
 
-        for resource_name, resource_stashcache_files in resources_stashcache_files.items():
-            for key, endpoint in keys_and_endpoints:
-                test_stashcache_file(key, endpoint, resources[resource_name]["FQDN"], resource_stashcache_files)
+            for resource_name, resource_stashcache_files in resources_stashcache_files.items():
+                for key, endpoint in keys_and_endpoints:
+                    test_stashcache_file(key, endpoint, resources[resource_name]["FQDN"], resource_stashcache_files)
+
+        finally:
+            if old_legacy_auth is None:
+                del app.config["STASHCACHE_LEGACY_AUTH"]
+            else:
+                app.config["STASHCACHE_LEGACY_AUTH"] = old_legacy_auth
 
     def test_stashcache_namespaces(self, client: flask.Flask):
         def validate_cache_schema(cc):
             assert HOST_PORT_RE.match(cc["auth_endpoint"])
             assert HOST_PORT_RE.match(cc["endpoint"])
-            assert HOST_PORT_RE.match(cc["resource"])
+            assert cc["resource"] and isinstance(cc["resource"], str)
```

Comment:
so i take it there is no longer any pattern requirement for cc["resource"] ?

Reply:
Doesn't one of the other CI tests check for the pattern? I don't want to duplicate that code.

Reply:
I'm not actually aware of any?  But sure if the check exists elsewhere (or if it should exist elsewhere) that's fine